### PR TITLE
Unify_certs: TRS changes and UT for new certificate scheme

### PR DESCRIPTION
**Pull request by situ-s**
* **Problem Overview**  
  Thin-Replica side changes for the new tls certificate scheme.
* **Testing Done**  
 Unit-tests are added for the changes.

### CODE REVIEW

File: client/clientservice/src/configuration.cpp [hunk 1]
```diff
@@ -206,11 +208,21 @@ void configureTransport(concord::client::concordclient::ConcordClientConfig& con
...
+        server_cert_path = tls_path + std::to_string(i) + "/node.cert";
+        std::string out_certs = "";
+        readCert(server_cert_path, out_certs);
+        config.transport.event_pem_certs += out_certs;
+      }
+    } else {
+      server_cert_path = tls_path + "/server.cert";
+      // read server TLS certs for this TRC instance
+      // server_cert_path specifies the path to a composite cert file i.e., a
+      // concatentation of the certificates of all known servers
```
Comment (yuliasherman) [inline client/clientservice/src/configuration.cpp:223]: concatentation->concatenation

File: tests/apollo/clientservice/config/clientservice.config [hunk 2]
```diff
@@ -29,6 +29,7 @@ tls_1_3_cipher_suite_list: TLS_AES_256_GCM_SHA384
 tls_1_3_comm_enabled: true
 tls_certificates_folder_path: /concord/tls_certs
 tls_cipher_suite_list: TLS_AES_256_GCM_SHA384
+use_unified_certificates: true
```
Comment (yuliasherman) [inline tests/apollo/clientservice/config/clientservice.config:32]: You cannot enable use_unified_certificates without enabling enable_multiplex_channel, too.

File: thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp [hunk 3]
```diff
@@ -39,10 +39,12 @@
 #include "thin_replica.grpc.pb.h"
 #include "subscription_buffer.hpp"
 #include "trs_metrics.hpp"
+#include <experimental/filesystem>
```
Comment (yuliasherman) [inline thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp:42]: Why do you need to use "experimental/filesystem"? ecursive_directory_iterator exists in "filesystem", too.

**Comment by robem** [comment]
@situ-s Is there a prior commit message that you can reference which explains what the "new certificate scheme" is? Or briefly mention it in this commit.

**Comment by situ-s** [comment]
> @situ-s Is there a prior commit message that you can reference which explains what the "new certificate scheme" is and why it was introduced? Or briefly mention it in this commit.

@robem , the new certificate scheme is being introduced for Lower end workloads. We will have common certificate scheme for Replica/TRS and Client/TRC.

**Pull request closed without merging**